### PR TITLE
Microsoft Graph Mail - additional commands

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
Ready

## Related Issues
related: https://github.com/demisto/etc/issues/19930

## Description
Implemented Microsoft Graph Mail news commands:
`msgraph-mail-list-folders`
`msgraph-mail-list-child-folders`
`msgraph-mail-create-folder`
`msgraph-mail-update-folder`
`msgraph-mail-delete-folder`
`msgraph-mail-move-email`
`msgraph-mail-get-email-as-eml`

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Dependencies
Test:
- [x] MicrosoftGraphMail-Test

## Technical writer review
Mention and link to the files that require a technical writer review.
- [x] [YAML file](https://github.com/demisto/content/blob/a3405228fadcde0d78fd74811c352972ae5423ab/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml)
- [x] [CHANGELOG](https://github.com/demisto/content/blob/a3405228fadcde0d78fd74811c352972ae5423ab/Integrations/MicrosoftGraphMail/CHANGELOG.md)

### CODE REVIEW

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 1]
```diff
@@ -178,7 +191,7 @@ def error_parser(resp_err: requests.Response) -> str:
 
 
 def http_request(method: str, url_suffix: str = '', params: dict = None, data: dict = None, odata: str = None,
-                 url: str = None) -> dict:
+                 url: str = None, resp_type: str = 'json', json_data: dict = None):
```
Comment (Itay4) [inline Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py]: why did you remove the return type hint?

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 2]
```diff
@@ -607,29 +635,182 @@ def list_attachments_command():
...
+
+def get_email_as_eml_command(args):
+    user_id = args.get('user_id')
+    message_id = args.get('message_id')
+
+    eml_content = get_email_as_eml(user_id, message_id)
+    file_result = fileResult(f'{message_id}.eml', eml_content)
+
+    if is_error(file_result):
+        demisto.error(file_result['Contents'])
```
Comment (Itay4) [inline Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py]: why is this needed? we already `return_error`

**Comment by yaakovi** [comment]
@kirbles19 please review CHANGELOG file and yml file (lines 58-59 and 350+)

**Comment by kirbles19** [comment]
@IkaDemisto Done.
